SportHub/src/modules/Header/components/Header.js

```diff
@@ -1,27 +1,36 @@
 import AdminHeader from "../../AdminHeader"
 import UserHeader from "../../UserHeader/components/UserHeader"
 import NotLoginedHeader from "../../NotLoginedHeader/componets/NotLoginedHeader"
-import {useAuthStore} from "../../../store/useAuthStore"
+import { useAuthStore } from "../../../store/useAuthStore"
+import { useEffect, useState } from "react"
+import { ROUTES } from "../../../routes/routes"
+import { useNavigate } from 'react-router-dom'
 
-export default function Header({setIsContentSearch, setContentSearchValue})
-{
+export default function Header() {
     const { userData } = useAuthStore()
+    const navigate = useNavigate()
+    const [isContentSearch, setIsContentSearch] = useState(false)
+    const [contentSearchValue, setContentSearchValue] = useState("")
+
+    useEffect(() => {
+        if (isContentSearch) {
+            setIsContentSearch(false)
+            navigate(ROUTES.SEARCH.replace(':searchValue', contentSearchValue))
+        }
+    }, [isContentSearch])
     return (
         <>
-            {(()=>{
-                if (userData?.isAdmin)
-                {
+            {(() => {
+                if (userData?.isAdmin) {
                     return <AdminHeader setIsContentSearch={setIsContentSearch} setContentSearchValue={setContentSearchValue} />
```

Comment:
make a store like I did with Categories in Header. So you won't need to pass it as a parameter to each header